src/features/ProfilePage/components/DisplayNameEditor.tsx

```diff
@@ -14,23 +15,25 @@ import { IconButton, TextButton } from '@/components/Buttons';
 import LabeledInput from '@/components/LabeledInput';
 import Text from '@/components/Text';
 
+import type { User } from '@/features/Authentication/authenticationApi';
+
 const DisplayNameEditor = () => {
   const { t } = useTranslation('profile');
-  const userInfo = useAppSelector(selectUserInfo);
+  const user = useAppSelector(selectUser);
+  const [updateUser, { isLoading, isSuccess }] = useUpdateUserMutation();
 
+  const [displayName, setDisplayName] = useState(user.display_name);
   const [isOpen, setIsOpen] = useState(false);
   const toggleIsOpen = () => setIsOpen(!isOpen);
-  const [displayName, setDisplayName] = useState('');
 
   const isTooShort = displayName.length < DISPLAY_NAME_MIN_LENGTH;
+  const isSavingDisabled = isLoading || isTooShort;
 
-  useEffect(() => {
-    setDisplayName(userInfo.displayName ?? '');
-  }, [userInfo.displayName]);
+  const userToSave: User = { ...user, display_name: displayName };
 
-  const saveNewDisplayName = () => {
-    console.log('API: Save new display name');
-  };
+  useEffect(() => {
+    if (isSuccess) setIsOpen(false);
+  }, [isSuccess]);
```

Comment:
This pattern seems to now triggered my radar. Not sure if I am just paranoid about too many useEffects or not. Anyway, I dont have a better solution for this in my mind.

Reply:
Let's proceed with this for now.